.ci/Jenkinsfile_cloud

```diff
@@ -19,51 +21,44 @@ pipeline {
-                                                        withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'password', variable_name: 'PASS_FROM_VAULT') {
-                                                            sh '''./kibana-load-testing/scripts/ingest_results.sh'''
-                                                        }
-                                                    }
-                                                }
-                                            }
-                                            break
-                                        } catch (Exception e) {
-                                            if (attempt == 3) {
-                                                throw new Exception("Could not succeed after ${attempt} attempts")
+                    for (def i = 1; i <= Integer.valueOf(env.NUMBER_DEPLOYMENTS); i++) {
+                        threads[i] = {
+                            docker.image(MAVEN_IMAGE).inside('-u root') {
+                                withVaultSecret(secret: 'secret/kibana-issues/dev/cloud-staging-api-key', secret_field: 'value', variable_name: 'API_KEY') {
+                                    sh """./kibana-load-testing/scripts/deploy_and_test.sh \
+                                        -v '${params.STACK_VERSION}' \
+                                        -c '${params.DEPLOY_CONFIG}' \
+                                        -s '${params.SIMULATION}' \
+                                        -n '${env.NUMBER_TEST_RUNS}'
+                                    """
+                                }
+                                loadPipeline.uploadGatlingReport()
+                                if (params.INGEST_RESULTS.toBoolean()) {
+                                    withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'host', variable_name: 'HOST_FROM_VAULT') {
+                                        withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'username', variable_name: 'USER_FROM_VAULT') {
+                                            withVaultSecret(secret: 'secret/kibana-issues/prod/coverage/elasticsearch', secret_field: 'password', variable_name: 'PASS_FROM_VAULT') {
+                                                sh '''./kibana-load-testing/scripts/ingest_results.sh'''
                                             }
                                         }
                                     }
                                 }
                             }
                         }
-                        // Still within the 'Script' block, run the parallel array object
-                        parallel threads
                     }
+                    // Still within the 'Script' block, run the parallel array object
+                    parallel threads
```

Comment:
Still waiting to properly test it. It might be a race condition with re-compiling the source code concurrently, then we need to make source copy in each thread.